src/vtcomposite.cpp

```diff
@@ -826,104 +824,130 @@ struct LocalizeWorker : Napi::AsyncWorker
-    }
-
-    Napi::Object buffer_obj = buf_val.As<Napi::Object>();
-    if (!buffer_obj.IsBuffer())
-    {
-        return utils::CallbackError("tile buffer is not a true buffer", info);
-    }
-
-    Napi::Buffer<char> buffer = buffer_obj.As<Napi::Buffer<char>>();
-
-    // validate language string
-    Napi::Value language_val = info[1];
+    // validate params object
+    Napi::Value params_val = info[0];
+    Napi::Buffer<char> buffer;
     std::string language;
     bool change_names = true;
-
-    if (language_val.IsNull())
+    std::vector<std::string> worldview;
+    std::string worldview_property = "_mbx_worldview";
+    bool compress = false;
+    if (!params_val.IsObject())
     {
-        change_names = false;
+        Napi::Error::New(info.Env(), "first argument must be an object").ThrowAsJavaScriptException();
+        return info.Env().Null();
     }
-    else if (!language_val.IsString())
+    Napi::Object params = info[0].As<Napi::Object>();
+
+    // params.buffer (required)
+    if (!params.Has(Napi::String::New(info.Env(), "buffer")))
     {
-        return utils::CallbackError("language value must be null or a string", info);
+        return utils::CallbackError("params.buffer is required", info);
     }
-    else
+    Napi::Value buffer_val = params.Get(Napi::String::New(info.Env(), "buffer"));
+    if (!buffer_val.IsObject() || buffer_val.IsNull() || buffer_val.IsUndefined())
     {
-        language = language_val.As<Napi::String>();
+        return utils::CallbackError("params.buffer must be a Buffer", info);
     }
 
-    if (change_names && language.length() == 0)
+    Napi::Object buffer_obj = buffer_val.As<Napi::Object>();
+    if (!buffer_obj.IsBuffer())
     {
-        return utils::CallbackError("language value is an empty string", info);
+        return utils::CallbackError("params.buffer is not a true Buffer", info);
     }
+    buffer = buffer_obj.As<Napi::Buffer<char>>();
 
-    // validate worldview string
-    Napi::Value worldview_val = info[2];
-    std::string worldview;
-
-    if (!worldview_val.IsString() && !worldview_val.IsNull())
+    // params.language (optional)
+    if (params.Has(Napi::String::New(info.Env(), "language")))
     {
-        return utils::CallbackError("worldview value must be null or a 2 character string", info);
+        Napi::Value language_val = params.Get(Napi::String::New(info.Env(), "language"));
+        if (!language_val.IsString() && !language_val.IsNull() && !language_val.IsUndefined())
+        {
+            return utils::CallbackError("params.language must be null or a string", info);
+        }
+        if (language_val.IsString())
+        {
+            language = language_val.As<Napi::String>();
+            if (language.length() == 0)
+            {
+                return utils::CallbackError("params.language cannot be an empty string", info);
+            }
+        }
+        if (language_val.IsNull() || language_val.IsUndefined())
```

Comment:
this could be an `else`. I think that would make the intent clearer here. Though, I am hoping the branch cann be removed entirely

Reply:
👍 agree we can clean this up - will leave any language validation changes for that PR though since it might all change.